### DISCUSSION TITLE
Enable caching in admin app when auto-purge is enabled

### DISCUSSION THREAD

**Discussion by daawaan4x** [directus/directus#15387] [Feature Requests]
Currently, the `Cache-Control` header in the axios instance of the admin app is set to `no-store` to ensure that all data fetched is in its most up-to-date state. When auto-purge is enabled however, I think the `Cache-Control` should be set to `no-cache` like how it was before it was set to `no-store` so caching can still be used as is in the admin app

**Comment by rijkvanzanten** [comment]
Auto-purge can't purge the browser's cache though, so even with auto-purge enabled, the browser cache will get stale causing all sorts of issues. As far as I can tell, `no-store` is the correct way, as it allows the API to internally cache and return early, while making sure the browser gets the updated payload whenever the API does eventually change

**Reply by daawaan4x** [reply]
Hey @rijkvanzanten, thanks for the reply. AFAIK, does `no-cache` not request a revalidation to the server on every request to check if the browser cache has already become stale? Which should consequently replace the browser cache on-demand I think. On a related note, I was wondering if the cache not working has anything to do with how the cache middleware immediately exits once it finds `no-store` (?)

https://github.com/directus/directus/blob/e7854549fdfd14f92f6dd1952fcb8a063cb0c0c9/api/src/middleware/cache.ts#L15-L20

Just for context, I was trying to set up caching in my directus instance but I wasn't noticing any difference in how long queries are being resolved even when there are no create/update/delete operations happening at the moment. Tried to enable the cache status header and all requests are getting `MISS`

**Reply by rijkvanzanten** [reply]
Ah from the app's perspective, that's correct and by design. The app will always request to circumvent the cache, to ensure you're managing the up-to-date data, as it currently doesn't know if auto-purge is enabled. I think the real improvement here would be to add a flag to the system info endpoint to tell the app that auto purging is enabled, and therefore caches don't have to be skipped (eg remove the no-store header from the app) 🙂

**Reply by daawaan4x** [reply]
Suggestion, would it be possible for the changes to just access the environment variables at the initialization of the axios instance for the admin app to decide whether 'Cache-Control' will be set to 'no-store' or not? 

Would also like to ask in the meantime, is there any way to override the default headers set with the axios instance being used at startup? I was thinking of `axios.default.headers.common['Cache-Control'] = 'no-cache'` but I can't think of a good place to put it

**Reply by rijkvanzanten** [reply]
No, the app can't read the environment variables, as that would mean that all private configuration is available client side. We're returning some of that information under the /server/info endpoint https://github.com/directus/directus/blob/e7854549fdfd14f92f6dd1952fcb8a063cb0c0c9/api/src/controllers/server.ts#L50-L62 for this type of use case, so "auto purge" could be one of the information pieces we return in there, which we can then use in dynamically updating the axios instance.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
